lib/timers/promises.js

```diff
@@ -111,7 +111,86 @@ function setImmediate(value, options = {}) {
       () => signal.removeEventListener('abort', oncancel)) : ret;
 }
 
+async function* setInterval(after, value, options = {}) {
+  if (options == null || typeof options !== 'object') {
+    throw new ERR_INVALID_ARG_TYPE(
+      'options',
+      'Object',
+      options);
+  }
+  const { signal, ref = true, throwOnAbort = true } = options;
+  validateAbortSignal(signal, 'options.signal');
+  if (typeof ref !== 'boolean') {
+    throw new ERR_INVALID_ARG_TYPE(
+      'options.ref',
+      'boolean',
+      ref);
+  }
+
+  if (typeof throwOnAbort !== 'boolean') {
+    throw new ERR_INVALID_ARG_TYPE(
+      'options.throwOnAbort',
+      'boolean',
+      ref);
+  }
+
+  if (signal?.aborted) {
+    if (throwOnAbort) throw new AbortError();
+    return;
+  }
+
+  let onCancel;
+  let notYielded = 0;
+  let passCallback;
+  let abortCallback;
+  const interval = new Timeout(() => {
+    notYielded++;
+    if (passCallback) {
+      passCallback();
+      passCallback = undefined;
+      abortCallback = undefined;
+    }
```

Comment:
Again, assigning the variables to `undefined` seems unnecessary 
```suggestion
    passCallback?.();
```

Reply:
I need the undefined there because otherwise we sometimes get an error during the tests because of multiple resolves